src/Regulator.php

```diff
@@ -489,8 +495,35 @@ protected function should_retry( Task $task ): bool {
 	 * Schedules the cleanup task.
 	 *
 	 * @since 0.0.1
+	 * @since 0.0.8 Updated to check if the Shepherd tables have been registered before scheduling the cleanup task.
 	 */
 	public function schedule_cleanup_task(): void {
-		$this->dispatch( new Herding(), 6 * HOUR_IN_SECONDS );
+		$prefix = Config::get_hook_prefix();
+
+		if ( ! did_action( "shepherd_{$prefix}_tables_registered" ) ) {
+			return;
+		}
+
+		/**
+		 * Filters whether to schedule the cleanup task.
+		 *
+		 * @since 0.0.8
+		 *
+		 * @param int $schedule_every_x_time The time in seconds to schedule the cleanup task. Default is 6 hours.
+		 */
+		$schedule_every_x_time = (int) apply_filters( "shepherd_{$prefix}_schedule_cleanup_task_every", 6 * HOUR_IN_SECONDS );
+
+		if ( 0 === $schedule_every_x_time ) {
+			return;
+		}
+
+		$this->dispatch( new Herding(), $schedule_every_x_time );
```

Comment:
From a previous installation of something using Shepherd, the action might have already been scheduled (do not think of a clean deactivation, think of a fatal error that prevented the plugin from deactivating properly). I would check whether the action is already scheduled and if so, do not schedule or deschedule it and reschdule it.

Reply:
This is something that is already taken care of as part of `dispatch`.

First it checks if there is the same task in AS, then even if not, the same task is checked against shepherd too.